### PR TITLE
chore(20379): Enable FileChannel api and DontSync sync option

### LINKED ISSUE #20379: PCES - enable `PcesFileWriterType.FILE_CHANNEL` and `FileSyncOption.DONT_SYNC`
After running a few programs [see](https://github.com/hiero-ledger/hiero-consensus-node/issues/18458)
and[experiments](https://github.com/hiero-ledger/hiero-consensus-node/pull/20048/files) 
where we test different scenarios simulating deterministic failures:

 - sig kill 
 - Runtime.getRuntime().halt(42), 
 - or by Unsafe.getUnsafe().putAddress(0L, 42L), 

That would cause the JVM to exit before calling the API that would produce the `fsync` system call in the OS., We confirmed that all the data was successfully stored in the file in all the scenarios that did not buffer.

The experiment is run as a JUnit test, and is [passing](https://scans.gradle.com/s/4mbylh2a32xyu/tests/task/:swirlds-platform-core:test/details/com.swirlds.platform.FileWritingCrashExperimentTest?top-execution=1) on GitHub runners too (also a Linux, if I understand correctly)

given that information, we want to enable the fileChannel writer for PCES and use the FileSyncOption.DONT_SYNC option measures iops and compare performance results.

### PR DESCRIPTION
**Pull request by mxtartaglia-sl**
**Description**:
After running a few programs https://github.com/hiero-ledger/hiero-consensus-node/issues/18458
and[experiments](https://github.com/hiero-ledger/hiero-consensus-node/pull/20048/files)
where we test different scenarios simulating deterministic failures:

sig kill
Runtime.getRuntime().halt(42),
or by Unsafe.getUnsafe().putAddress(0L, 42L),
That would cause the JVM to exit before calling the API that would produce the fsync system call in the OS., We confirmed that all the data was successfully stored in the file in all the scenarios that did not buffer.

The experiment is run as a JUnit test, and is [passing](https://scans.gradle.com/s/4mbylh2a32xyu/tests/task/:swirlds-platform-core:test/details/com.swirlds.platform.FileWritingCrashExperimentTest?top-execution=1) on GitHub runners too (also a Linux, if I understand correctly)

given that information, we want to enable the fileChannel writer for PCES and use the FileSyncOption.DONT_SYNC option measures iops and compare performance results.

**Related issue(s)**:

Fixes #20379

### CODE REVIEW

**Comment by alex-kuzmin-hg** [comment]
@mxtartaglia-sl
Benchmarks: https://perf.analytics.eng.hashgraph.io/ephemeral/test_runs/20379-enable-filechannel-dontsync_AdHoc5_170/report/ 
AsynProf:
https://perf.analytics.eng.hashgraph.io/ephemeral/AsyncProfiler/AdHoc5_75_16:42:31:07-30-2025/report/

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesFileChannelWriter.java [hunk 1]
```diff
@@ -65,10 +65,10 @@ public void writeVersion(final int version) throws IOException {
     @Override
     public long writeEvent(@NonNull final GossipEvent event) throws IOException {
         final int size = GossipEvent.PROTOBUF.measureRecord(event);
-        final boolean expandBuffer = size > buffer.capacity();
+        final boolean expandBuffer = (size + Integer.BYTES) > buffer.capacity();
         if (expandBuffer) {
             MemoryUtils.closeDirectByteBuffer(buffer);
-            buffer = ByteBuffer.allocateDirect(size);
+            buffer = ByteBuffer.allocateDirect(size + Integer.BYTES);
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesFileChannelWriter.java:71]: is there a unit test that tests this edge case?

Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesFileChannelWriter.java:71]: I'll create a ticket for this
